### ISSUE TITLE
Improve the accessibility of language drop-down

### ISSUE THREAD

**Issue by schalkneethling** [mozilla/kuma#5859]
**Summary**

From https://bugzilla.mozilla.org/show_bug.cgi?id=1583146 reported by @marcozehe

## What did you do?

1. With NVDA on Windows, visit any MDN documentation page.
2. Find the heading 1, and down arrow a few lines until you get to the button that speaks your language as the name, e. g. "English".
3. DownArrow once more. You'll hear "button" spoken again. But there is no label.
4. Try to open the button menu with the keyboard. 

### What happened?

After step 3, No label was spoken, and the button does not indicate that it will open a popup for a language selection.
After step 4, the menu isn't keyboard operable, only drops down on mouse hover.

## What should have happened?

There should not be an empty portion of the button, and it should indicate that it has a popup. The menu should be keyboard operable, even without a screen reader.

**Additional context**

The solution is probably to make the span and its children that shows the dropdown arrow aria-hidden="true", since this is purely presentational. The button accessible doesn't need this, since it laready has the language name as its label.

Second, it should have aria-haspopup="menu" or similarly appropriate, to indicate that there is a menu attached toit. And because this is initially collapsed, it should also have aria-expanded="false".. The latter, of course, changes to "true" once the button gets expanded, and the menu appears.

And of course, this must be made keyboard accessible. It's a menu, like other menus on MDN, too, so the same user interaction applies, like focusing the first menu item when expanding with the keyboard, up and down arrow selects, escape closes and puts focus back on the language button, etc. The list is a menu, the li is presentation, the links are menuitem roles.

**Comment by Gregoor** [comment]
Out of curiosity, why aren't we using a regular ol' `<select>` for this?

**Comment by schalkneethling** [comment]
> Out of curiosity, why aren't we using a regular ol' `<select>` for this?

I guess it was an aesthetics decision. Styling a `select` is cumbersome and an especially hard challenge cross-browser. More often than not, once you start down that road you end up where we are now ;)

With that said, I am would not be opposed to just using a plain ol' `select` element without any fancy styling.

**Comment by Gregoor** [comment]
Yeah I'd definitely vote for that, since styling the dropdown doesn't seem as pressing to me as accessibility and (relatedly) keyboard navigtability, including search. Which both do take some time to implement and get right (awesome lib for that btw: https://github.com/downshift-js/downshift 11kb though).
